peg.rb

```diff
@@ -0,0 +1,61 @@
+require_relative 'constants'
+
+class Peg
+	def initialize(peg_line, num_pegs = 4)
+		@peg_line = peg_line
+		@num_pegs = num_pegs
+	end
+
+	def calculate_closeness(codeline)
```

Comment:
You are using more variables than necessary and 3 loops which is bad practice. You can refactor this method to reduce its complexity.